src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -73,28 +76,72 @@ private boolean isReady(BlobItem blob, String containerName) {
     }
 
     private void processBlob(BlobClient blob) {
-        envelopeService
-            .findLastEnvelope(blob.getBlobName(), blob.getContainerName())
+        getLastEnvelope(blob)
             .ifPresentOrElse(
                 envelope -> {
                     if (envelope.status == Status.CREATED) {
-                        leaseAndThen(blob, () -> blobProcessor.continueProcessing(envelope.id, blob));
+                        leaseAndThen(blob, () ->
+                            continueProcessingEnvelopeIfItStillExists(
+                                blob,
+                                this::logEnvelopeAlreadyDeleted
+                            )
+                        );
                     } else {
-                        logger.info("Envelope already processed in system, skipping. {} ", envelope.getBasicInfo());
+                        logEnvelopeAlreadyProcessed(envelope);
                     }
                 },
-                () -> leaseAndThen(blob, () -> blobProcessor.process(blob))
+                () -> leaseAndThen(blob, () ->
+                    continueProcessingEnvelopeIfItStillExists(
+                        blob,
+                        blobProcessor::process
+                    )
+                )
             );
     }
 
-    private void leaseAndThen(BlobClient blobClient, Runnable action) {
+    private void continueProcessingEnvelopeIfItStillExists(
+        BlobClient blob,
+        Consumer<BlobClient> nonExistingEnvelopeHandler
+    ) {
+        getLastEnvelope(blob)
+            .ifPresentOrElse(
+                envelopePotentiallyProcessed -> continueProcessingIfPossible(blob, envelopePotentiallyProcessed),
+                () -> nonExistingEnvelopeHandler.accept(blob)
+            );
+    }
+
+    private void continueProcessingIfPossible(BlobClient blob, Envelope envelope) {
+        if (envelope.status == Status.CREATED) {
+            blobProcessor.continueProcessing(envelope.id, blob);
+        } else {
+            logEnvelopeAlreadyProcessed(envelope);
+        }
+    }
+
+    private void logEnvelopeAlreadyDeleted(BlobClient blob) {
+        logger.error(
+            "Envelope already deleted in system for blob {}, container {}",
+            blob.getBlobName(), blob.getContainerName()
+        );
+    }
```

Comment:
Suggestion: maybe it would be best to get rid of `already` from the names and message? We typically use already when there's no need to do something, because it has already happened. This, however, is an error scenario -envelopes should never be deleted.